test/metabase/api/action_test.clj

```diff
@@ -588,3 +590,39 @@
             (is (partial= {:message "No destination parameter found for #{\"name\"}. Found: #{\"last_login\" \"id\"}"}
                           (mt/user-http-request :crowberto :post 400 (format "action/%s/execute" action-id)
                                                 {:parameters {:name "Darth Vader"}})))))))))
+
+(deftest fetch-implicit-action-default-values-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :actions)
+    (mt/with-actions-enabled
+      (mt/with-actions [_                             {:dataset true :dataset_query (mt/mbql-query venues {:fields [$id $name]})
+                                                       :collection_id (:id (collection/user->personal-collection (mt/user->id :crowberto)))}
+                        {create-action-id :action-id} {:type :implicit :kind "row/create"}
+                        {update-action-id :action-id} {:type :implicit :kind "row/update"}
+                        {delete-action-id :action-id} {:type :implicit :kind "row/delete"}
+                        {http-action-id :action-id}   {:type :http}
+                        {query-action-id :action-id}  {:type :query}]
+        (testing "403 if user does not have permission to view the action"
+          (is (= "You don't have permissions to do that."
+                 (mt/user-http-request :rasta :get 403 (format "action/%d/execute" update-action-id) :parameters (json/encode {:id 1})))))
+
+        (testing "404 if id does not exists"
+          (is (= "Not found."
+                 (mt/user-http-request :rasta :get 404 (format "action/%d/execute" Integer/MAX_VALUE) :parameters (json/encode {:id 1})))))
+
+        (testing "returns empty map for actions that are not implicit"
+          (is (= {}
+                 (mt/user-http-request :crowberto :get 200 (format "action/%d/execute" http-action-id) :parameters (json/encode {:id 1}))))
+          (is (= {}
+                 (mt/user-http-request :crowberto :get 200 (format "action/%d/execute" query-action-id) :parameters (json/encode {:id 1})))))
+
+        (testing "Can't fetch for create action"
+          (is (= "Values can only be fetched for actions that require a Primary Key."
+                 (mt/user-http-request :crowberto :get 400 (format "action/%d/execute" create-action-id) :parameters (json/encode {:id 1})))))
+
+        (testing "fetch for update action return name and id"
+          (is (= {:id 1 :name "Red Medicine"}
+                 (mt/user-http-request :crowberto :get 200 (format "action/%d/execute" update-action-id) :parameters (json/encode {:id 1})))))
+
```

Comment:
There should be a test checking that fetching values doesn't work if actions are disabled for the database. You can use `mt/with-actions-disabled`